compiler/pragmas.nim

```diff
@@ -691,14 +691,18 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: int,
   let ident = considerQuotedIdent(key)
   var userPragma = strTableGet(c.userPragmas, ident)
   if userPragma != nil:
-    inc c.instCounter
-    if c.instCounter > 100:
-      globalError(it.info, errRecursiveDependencyX, userPragma.name.s)
-    pragma(c, sym, userPragma.ast, validPragmas)
-    # ensure the pragma is also remember for generic instantiations in other
-    # modules:
-    n.sons[i] = userPragma.ast
-    dec c.instCounter
+    if userPragma.ast.len == 0:
+      delete(n.sons, i)
```

Comment:
Why?

Reply:
If user pragma is ended-up being empty (not uncommon situation) after expansion of user pragma you will get one pragma less: 
Before expansion: ``proc x {.my_pragma, inline.}``
After expansion: ``proc x {.inline.}``

P.S.
I have found a way to merge `` n.sons[i..i] = userPragma.ast.sons`` and ``delete(n.sons, i)`` branches into one. I will submit an updated version